File: app/src/main/java/com/xeniac/warrantyroster/LocaleModifier.java
package com.xeniac.warrantyroster;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.util.DisplayMetrics;

import java.util.Locale;

public class LocaleModifier {

    private final Context mContext;

    public LocaleModifier(Context context) {
        this.mContext = context;
    }

    public void setLocale() {
        SharedPreferences localePrefs = mContext.getSharedPreferences(Constants.PREFERENCE_SETTINGS, Context.MODE_PRIVATE);
        String language = localePrefs.getString(Constants.PREFERENCE_LANGUAGE_KEY, "en");
        String country = localePrefs.getString(Constants.PREFERENCE_COUNTRY_KEY, "US");

        Resources resources = mContext.getResources();
        DisplayMetrics displayMetrics = resources.getDisplayMetrics();
        Configuration configuration = resources.getConfiguration();
        Locale newLocale = new Locale(language, country);
        Locale.setDefault(newLocale);
        configuration.setLocale(newLocale);
        resources.updateConfiguration(configuration, displayMetrics);
    }
}

File: app/src/main/java/com/xeniac/warrantyroster_manager/utils/LocaleModifier.kt
package com.xeniac.warrantyroster_manager.utils

import android.content.Context
import java.util.*

//TODO REMOVE
object LocaleModifier {

//    private lateinit var currentLanguage: String
//    private lateinit var currentCountry: String

//    @EntryPoint
//    @InstallIn(SingletonComponent::class)
//    interface CurrentLanguageProviderEntryPoint {
//        @CurrentLanguage
//        fun getCurrentLanguage(): String
//    }
//
//    @EntryPoint
//    @InstallIn(SingletonComponent::class)
//    interface CurrentCountryProviderEntryPoint {
//        @CurrentCountry
//        fun getCurrentCountry(): String
//    }

    fun setLocale(context: Context) {
//        val languageProviderEntryPoint = EntryPointAccessors
//            .fromApplication(context, CurrentLanguageProviderEntryPoint::class.java)
//        val countryProviderEntryPoint = EntryPointAccessors
//            .fromApplication(context, CurrentCountryProviderEntryPoint::class.java)

//        currentLanguage = languageProviderEntryPoint.getCurrentLanguage()
//        currentCountry = countryProviderEntryPoint.getCurrentCountry()

        val currentLanguage = "en"
        val currentCountry = "US"

        val resources = context.resources
        val displayMetrics = resources.displayMetrics
        val configuration = resources.configuration
        val newLocale = Locale(currentLanguage, currentCountry)
        Locale.setDefault(newLocale)
        configuration.setLocale(newLocale)
        resources.updateConfiguration(configuration, displayMetrics)
    }
}